### PR TITLE
fix(db): the function UpdateProvingStatusFailed proving_status determines the condition incorrectly

### PR DESCRIPTION
**Pull request by Hsiao-Jan**
### Purpose or design rationale of this PR

*Describe your change. Make sure to answer these three questions: What does this PR do? Why does it do it? How does it do it?*

This PR fixes the problem that the UpdateProvingStatusFailed function incorrectly uses `type.ProverProofValid` as the filtering condition when using proving_status as the judging condition.

### PR title

Your PR title must follow [[conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [[types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] fix: A bug fix

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [ ] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

**Comment by georgehao** [comment]
make nosense

**Comment by georgehao** [comment]
thx for your pr, good job.

**Review by georgehao — DISMISSED** [review]

File: go.work.sum
Comment (Thegaram) [inline go.work.sum:1]: These changes seem irrelevant, can you undo them please?

**Review by georgehao — DISMISSED** [review]

**Review by 0xmountaintop — APPROVED** [review]

**Review by georgehao — APPROVED** [review]
